### PR TITLE
Minor fix - enable UTF-32 character key property to an object

### PR DESCRIPTION
**Pull request by kisbg**
Allows UTF-32 characters in identifiers.

Signed-off-by: bence gabor kis <kisbg@inf.u-szeged.hu>

### CODE REVIEW

**Comment by kisbg** [comment]
for example:

```
var a = {$𐒤 : 4 }
print(a.$𐒤)
```

**Comment by clover2123** [comment]
Could you enable test262 TCs which are related to this pr?
You can update the `exclude list` (tools/test/test262/excludelist.orig.xml) by running `make_excludelist.py` script(tools/test/test262/make_excludelist.py)
```sh
./tools/test/test262/make_excludelist.py --arch=x86 --engine="./out_linux_release/escargot" (default: --arch='x86_64' --engine='PROJECT_ROOT/escargot')
```

**Comment by kisbg** [comment]
@clover2123  I used the make_excludelist.py, but there isn't a specific test-cases that uses utf-32 character identifier

File: third_party/yarr/YarrParser.h [hunk 1]
```diff
@@ -1126,15 +1126,34 @@ class Parser {
...
+        UChar32 ch = tryConsumeIdentifierCharacter();
+        if (U16_IS_LEAD(ch) && m_isUnicode && (patternRemaining() > 0)) {
+            ParseState state = saveState();
+
+                      UChar32 surrogate2 = consume();
+                      if (U16_IS_TRAIL(surrogate2))
+                          ch = U16_GET_SUPPLEMENTARY(ch, surrogate2);
+                      else
+                          restoreState(state);
+                  }
```
Comment (clover2123) [inline third_party/yarr/YarrParser.h]: Indentation of this part is broken. Please fix it.

File: third_party/yarr/YarrParser.h [hunk 2]
```diff
@@ -1126,15 +1126,34 @@ class Parser {
...
+                UChar32 ch = tryConsumeIdentifierCharacter();
+                if (U16_IS_LEAD(ch) && m_isUnicode && (patternRemaining() > 0)) {
+                    ParseState state = saveState();
+
+                              UChar32 surrogate2 = consume();
+                              if (U16_IS_TRAIL(surrogate2))
+                                  ch = U16_GET_SUPPLEMENTARY(ch, surrogate2);
+                              else
+                                  restoreState(state);
+                          }
```
Comment (clover2123) [inline third_party/yarr/YarrParser.h]: Please check the indentation here too. Other looks fine to me:)

**Pull request merged**